### DISCUSSION TITLE
Difference b/w @@id and @@unique.

### DISCUSSION THREAD

**Discussion by curiosbasant** [prisma/prisma#14341] [Q&A]
```
model User {
  id        String    @id @default(cuid())
  createdAt DateTime  @default(now())
  updatedAt DateTime? @updatedAt

  username  String  @unique
  email     String  @unique
  password  String
  avatarUrl String?
  coverUrl  String?

  connectionRequests Connection[]      @relation("Requestor")
  connections        Connection[]      @relation("Acceptor")
}

model Connection {
  assignedAt DateTime @default(now())
  accepted   Boolean  @default(false)

  requestor   User   @relation("Requestor", fields: [requestorId], references: [id])
  requestorId String
  acceptor    User   @relation("Acceptor", fields: [acceptorId], references: [id])
  acceptorId  String

  @@id([requestorId, acceptorId])  // 1
  @@unique([requestorId, acceptorId])  // 2
}
```

In this case what is the difference between using `@@id` and `@@unique`. 
Which one should I use?

**Comment by 3xau1o** [comment]
you must have only one id, on the other side you can have any number of unique columns

**Comment by well-balanced** [comment]
id(pk) is clustered index and unique index is non clustered index. you can check differences between each other from [here](https://www.geeksforgeeks.org/difference-between-clustered-and-non-clustered-index/)

in this case, `@@unique` is better.

it is more appropriate to use a monotonically increasing id for the primary key. If there are more rows, there could be performance issues when working on the insert. your composite id(requestorId & acceptorId) is not a monotonically increasing id.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
